packages/legend-query/src/components/QueryBuilder.tsx

```diff
@@ -106,6 +115,8 @@ export const QueryBuilder = observer(
     const { queryBuilderState } = props;
     const applicationStore = useApplicationStore();
     const isQuerySupported = queryBuilderState.isQuerySupported();
+    const [showFunctionPanel, setShowFunctionPanel] = useState(false);
```

Comment:
why do we use local states here? For these, we could add states to the `QueryBuilderState` I think.